src/sentry/static/sentry/app/views/dashboardsV2/widgetQueries.tsx

```diff
@@ -120,21 +127,67 @@ class WidgetQueries extends React.Component<Props, State> {
   async fetchData() {
     const {selection, api, organization, widget} = this.props;
 
-    this.setState({loading: true, results: []});
+    this.setState({loading: true});
 
-    const statsPeriod = selection.datetime.period;
     const {start, end} = selection.datetime;
     const {projects, environments} = selection;
-    const interval = getWidgetInterval(widget.interval, {
-      start,
-      end,
-      period: statsPeriod,
-    });
-
-    const promises: Promise<EventsStats | MultiSeriesEventsStats>[] = widget.queries.map(
-      query => {
-        // TODO(mark) adapt this based on the type of widget being built.
-        // Table and stats results will need to do a different request.
+
+    if (widget.displayType === 'table') {
+      // Table and stat widgets use table results and need
+      // to do a discover 'table' query instead of a 'timeseries' query.
+      this.setState({tableResults: []});
+      const promises = widget.queries.map(query => {
+        const eventView = EventView.fromSavedQuery({
+          id: undefined,
+          name: query.name,
+          version: 2,
+          fields: query.fields,
+          query: query.conditions,
+          projects,
+          start: start ? getUtcDateString(start) : undefined,
+          end: end ? getUtcDateString(end) : undefined,
+        });
+        const url = `/organizations/${organization.slug}/eventsv2/`;
+        return doDiscoverQuery<TableData>(api, url, {
+          ...eventView.generateQueryStringObject(),
+          per_page: 5,
+        });
+      });
+
+      let completed = 0;
+      promises.forEach(async (promise, i) => {
+        try {
+          const result = await promise;
+          // Cast so we can add a title.
+          const tableData = result[0] as TableDataWithTitle;
+          tableData.title = widget.queries[i]?.name ?? '';
+
+          completed++;
+          this.setState(prevState => {
+            const tableResults = [...prevState.tableResults, tableData];
+            return {
+              ...prevState,
+              tableResults,
+              errorMessage: undefined,
```

Comment:
If, for example, there were two queries here. And the first one failed while the second one succeeded. Wouldn't this hide the error because it only contains the `errorMessage` value from the last request?